spec/models/event_subscriber_spec.rb

```diff
@@ -18,7 +18,17 @@
     it 'EventSubscriber does not send event when book is central' do
       book.switch!
       expect_any_instance_of(EventSubscriber).not_to receive(:do_request)
-      EventSubscriber.notify_sync!(Event::Submission.new(assignment))
+      EventSubscriber.notify!(Event::Submission.new(assignment))
+    end
+  end
+
+  describe 'test event subscriber api' do
+    before { expect_any_instance_of(EventSubscriber).to receive(:do_request).and_return({'status' => 'ok'}.to_json) }
+    let(:user) { create(:user) }
+    let(:exercise) { create(:x_equal_5_exercise) }
+
+    it 'run notifier when submit a solution' do
+      exercise.submit_solution! user
     end
```

Comment:
I know you can not add any assert here since there is nothing to assert - just that it doesn't fail - and there is also an expectation set in the before block. 

But just to keep intention clear, I would write it with an `expect { ... }.to_not raise_error`